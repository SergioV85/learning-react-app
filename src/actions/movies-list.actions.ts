import * as MoviesListActionTypes from './../action-types/movies-list.action-types';
import { getMoviesList } from './../services/movies.service';

const searchMovies = () => {
  return (dispatch: any, getState: any) => {
    const { searchParams } = getState();
    dispatch({
      payload: searchParams,
      type: MoviesListActionTypes.MoviesListActionTypes.GetMovies,
    });

    return getMoviesList(searchParams)
      .then((list) => dispatch({
        payload: list.data,
        type: MoviesListActionTypes.MoviesListActionTypes.GetMoviesComplete,
      }))
      .catch((err) => dispatch({
        payload: err,
        type: MoviesListActionTypes.MoviesListActionTypes.GetMoviesError,
      }));
  };
};

export const MoviesListActions = {
  searchMovies,
};
