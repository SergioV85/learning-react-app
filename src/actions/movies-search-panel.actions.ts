import * as MoviesSearchPanelActionTypes from './../action-types/movies-search-panel.action-types';

const updateKeyword = (keyword: string) => ({
  payload: keyword,
  type: MoviesSearchPanelActionTypes.MoviesSearchPanelActionTypes.UpdateSearchKeyword,
});
const updateSortingOrder = (sortOrder: string) => ({
  payload: sortOrder,
  type: MoviesSearchPanelActionTypes.MoviesSearchPanelActionTypes.UpdateSortOrder,
});
const updateSortingType = (sortType: string) => ({
  payload: sortType,
  type: MoviesSearchPanelActionTypes.MoviesSearchPanelActionTypes.UpdateSortType,
});
const updateType = (searchType: string) => ({
  payload: searchType,
  type: MoviesSearchPanelActionTypes.MoviesSearchPanelActionTypes.UpdateSearchType,
});

export const MoviesSearchPanelActions = {
  updateKeyword,
  updateSortingOrder,
  updateSortingType,
  updateType,
};
