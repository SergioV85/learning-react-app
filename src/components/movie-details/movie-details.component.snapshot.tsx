import { shallow } from 'enzyme';
import * as React from 'react';
import { mockedMovieDetails } from './../movies-list/mocked-details';
import { MovieDetails } from './movie-details.component';

describe('Movie Details Component Snapshot', () => {
  it('should renders correctly', () => {
    const movieDetailsSnapshot = shallow(
      <MovieDetails selectedMovie={mockedMovieDetails} />,
    );
    expect(movieDetailsSnapshot)
      .toMatchSnapshot();
  });
});
