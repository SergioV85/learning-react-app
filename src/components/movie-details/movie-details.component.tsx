import * as React from 'react';
import { DateTime } from 'luxon';
import { MovieObject } from './../movie-card/movie-card.component';

interface MovieDetailsProps {
  selectedMovie: MovieObject
}

interface MovieDetailsState {}

export class MovieDetails extends React.Component<MovieDetailsProps, MovieDetailsState> {
  constructor(props: MovieDetailsProps) {
    super(props);
  }

  public render() {
    return <div className="c-movie-details">
      <img className="c-movie-details__poster" src={this.props.selectedMovie.poster_path} />
      <div className="c-movie-details__info">
        <h4 className="c-movie-details__header">
          <span className="c-movie-details__title">{this.props.selectedMovie.title}</span>
          <span className="c-movie-details__rating">{this.props.selectedMovie.vote_average}</span>
        </h4>
        <p className="c-movie-details__tagline">
          {this.props.selectedMovie.tagline}
        </p>
        <p>
          <span className="c-movie-details__year">
            { DateTime.fromISO(this.props.selectedMovie.release_date).year }
          </span>
          <span className="c-movie-details__duration">
            {this.props.selectedMovie.runtime} min
          </span>
        </p>
        <div className="c-movie-details__description">
          {this.props.selectedMovie.overview}
        </div>
      </div>
    </div>;
  }
}