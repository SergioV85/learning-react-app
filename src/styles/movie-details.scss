.c-movie-details {
  display: flex;
  flex-direction: row;
  margin: 30px 0;
  width: 100%;
  
  &__poster {
    margin-right: 20px;
    max-height: 250px;
    width: auto;
  }
  &__info {
    flex: 1 1 auto;
  }
  &__header {
    display: flex;
  }
  &__title {
    color: $light-red;
    font-family: $tilte-font;
    font-size: 1.2rem;
    font-weight: 700;
    margin-right: 10px;
  }
  &__rating {
    align-items: center;
    border: 1px solid $primary-green;
    border-radius: 100%;
    color: $light-green;
    display: flex;
    font-family: $tilte-font;
    font-size: 0.9rem;
    height: 24px;
    justify-content: center;
    width: 24px;
  }
  &__tagline {
    color: $primary-grey;
    font-family: $main-font;
    font-size: 0.9rem;
  }
  &__year {
    color: $primary-light;
    font-family: $main-font;
    font-size: 1rem;
    margin-right: 20px;
  }
  &__duration {
    color: $primary-light;
    font-family: $main-font;
    font-size: 1rem;
  }
  &__description {
    color: $primary-grey;
    font-family: $main-font;
    font-size: 0.9rem;
    margin-top: 20px;
  }
}