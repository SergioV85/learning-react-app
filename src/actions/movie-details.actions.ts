import * as MovieDetailsActionTypes from './../action-types/movie-details.action-types';
import { getMovieDetails } from './../services/movies.service';

const backToListAction = () => ({
  type: MovieDetailsActionTypes.MovieDetailsActionTypes.BackToList,
});
const getMovie = (selectedMovieId: number) => {
  return (dispatch: any) => {
    dispatch({
      payload: selectedMovieId,
      type: MovieDetailsActionTypes.MovieDetailsActionTypes.GetMovie,
    });

    return getMovieDetails(selectedMovieId)
      .then((details) => dispatch({
        payload: details.data,
        type: MovieDetailsActionTypes.MovieDetailsActionTypes.GetMovieComplete,
      }))
      .catch((err) => dispatch({
        payload: err,
        type: MovieDetailsActionTypes.MovieDetailsActionTypes.GetMovieError,
      }));
  };
};

export const MovieDetailsActions = {
  backToListAction,
  getMovie,
};
