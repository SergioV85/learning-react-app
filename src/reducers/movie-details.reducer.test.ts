import { MovieDetailsActions } from './../actions/movie-details.actions';
import { movieDetailsReducer } from './movie-details.reducer';

const mockedState = {
  isLoading: false,
};

const initialState = movieDetailsReducer(mockedState, {} as any);

describe('Movie Details Reducer', () => {
  describe('initial state', () => {
    it('should match a snapshot', () => {
      expect(initialState).toMatchSnapshot();
    });
  });
});
