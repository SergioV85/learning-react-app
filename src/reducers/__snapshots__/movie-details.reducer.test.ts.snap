// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details Reducer initial state should match a snapshot 1`] = `
Object {
  "isLoading": false,
}
`;
